src/main/java/skywalker/parser/Parser.java

```diff
@@ -0,0 +1,124 @@
+package skywalker.parser;
+
+import skywalker.command.AddCommand;
+import skywalker.command.Command;
+import skywalker.command.DeleteCommand;
+import skywalker.command.ExitCommand;
+import skywalker.command.FindCommand;
+import skywalker.command.ListCommand;
+import skywalker.command.MarkCommand;
+import skywalker.task.Deadline;
+import skywalker.task.Event;
+import skywalker.task.Task;
+import skywalker.task.Todo;
+
+/**
+ * The Parser class is reponsible of receiving the user's input
+ * and to convert it to commands or tasks that can be executed by the app.
+ */
+public class Parser {
+
+    /**
+     * This method looks into what type of command the user input
+     * and then processes the remaining part of the input
+     * @param fullCommand The string inputted by the user
+     * @return The command object corresponding to the users' input
+     * @throws Exception If the command is unknown/invalid.
+     */
+    public static Command parse(String fullCommand) throws Exception {
```

Comment:
Perhaps a more descriptive method name such as parseCommand would be better?

Reply:
I think so too, since there is another method starting with parse as well